presto-main/src/test/java/com/facebook/presto/operator/scalar/TestHyperLogLogFunctions.java

```diff
@@ -0,0 +1,181 @@
+package com.facebook.presto.operator.scalar;
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.facebook.presto.Session;
+import com.facebook.presto.SessionTestUtils;
+import com.facebook.presto.sql.analyzer.FeaturesConfig;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.testing.MaterializedResult;
+
+import io.airlift.slice.Slice;
+import io.airlift.stats.cardinality.HyperLogLog;
+
+import org.testng.annotations.Test;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+
+import com.google.common.io.BaseEncoding;
+
+
+public class TestHyperLogLogFunctions
+        extends AbstractTestFunctions
+{
+    private TestHyperLogLogFunctions() {}
+
+    private Long getCardinality(String projection)
+    {
+        //variables from FunctionAssertions.java to get results of query
+        Session session = SessionTestUtils.TEST_SESSION;
+        FeaturesConfig featuresConfig = new FeaturesConfig();
+        LocalQueryRunner runner = new LocalQueryRunner(session, featuresConfig);
+
+        MaterializedResult result = runner.execute("SELECT " + projection);
+        //will return a Long, since that's the return type for cardinality()
+        Long cardinality = (Long) result.getOnlyValue();
+
+        return cardinality;
+    }
+
+    //Description: writes string that will be used as projection, for calculating cardinality given an array of HyperLogLogs
```

Comment:
Once you refactor this, you may remove the comment as it should be obvious what's happening.